src/parser.rs

```diff
@@ -3588,5 +3628,47 @@ mod test {
         } else {
           panic!("Failed to read fake env float");
         }
+
```

Comment:
I use the library in my project. So thank you.
You merged it already but I have an idea on error handling. As you know auto type injector can be written in two ways the short form `$"SOME_VAR_NAME"` and the long form `$"SOME_VAR_NAME"::auto`.
My idea is to make the short version permissive, when there is no env var it returns an empty string, and to make the long form restrictive, when there is no env var it returns an error.